File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/context/julia/runtime/contract/InputsAndResultsTestInteger.jl
println("getInteger")
if isdefined(Main, :getInteger) && !isnothing(getInteger)
	error("getInteger already defined!")
end
getInteger = putInteger
println(typeof(getInteger))
println(getInteger)
if typeof(getInteger) != Int32
	error("getInteger not Int32!")
end

println("getIntegerVector")
if isdefined(Main, :getIntegerVector) && !isnothing(getIntegerVector)
	error("getIntegerVector already defined!")
end
getIntegerVector = putIntegerVector
println(eltype(getIntegerVector))
println(getIntegerVector)
if eltype(getIntegerVector) != Int32
	error("getIntegerVector not Int32!")
end

println("getIntegerVectorAsList")
if isdefined(Main, :getIntegerVectorAsList) && !isnothing(getIntegerVectorAsList)
	error("getIntegerVectorAsList already defined!")
end
getIntegerVectorAsList = putIntegerVectorAsList
println(eltype(getIntegerVectorAsList))
println(getIntegerVectorAsList)
if eltype(getIntegerVectorAsList) != Int32
	error("getIntegerVectorAsList not Int32!")
end

println("getIntegerMatrix")
if isdefined(Main, :getIntegerMatrix) && !isnothing(getIntegerMatrix)
	error("getIntegerMatrix already defined!")
end
getIntegerMatrix = putIntegerMatrix
println(eltype(getIntegerMatrix))
println(getIntegerMatrix)
if eltype(getIntegerMatrix) != Int32
	error("getIntegerMatrix not Int32!")
end

println("getIntegerMatrixAsList")
if isdefined(Main, :getIntegerMatrixAsList) && !isnothing(getIntegerMatrixAsList)
	error("getIntegerMatrixAsList already defined!")
end
getIntegerMatrixAsList = putIntegerMatrixAsList
println(eltype(getIntegerMatrixAsList))
println(getIntegerMatrixAsList)
if eltype(getIntegerMatrixAsList) != Int32
	error("getIntegerMatrixAsList not Int32!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestInteger.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getInteger")
if(type(getInteger) ~= "nil") then
	error("getInteger already defined!")
end
getInteger = putInteger
getIntegerType = type(getInteger)
print(getIntegerType)
print(getInteger)
if(getIntegerType ~= "number") then
	error("getInteger not Integer!")
end

print("getIntegerVector")
if(type(getIntegerVector) ~= "nil") then
	error("getIntegerVector already defined!")
end
getIntegerVector = putIntegerVector
getIntegerVectorType = type(getIntegerVector[1])
print(getIntegerVectorType)
print(arraysToString(getIntegerVector))
if(getIntegerVectorType ~= "number") then
	error("getIntegerVector not Integer!")
end

print("getIntegerVectorAsList")
if(type(getIntegerVectorAsList) ~= "nil") then
	error("getIntegerVectorAsList already defined!")
end
getIntegerVectorAsList = putIntegerVectorAsList
getIntegerVectorAsListType = type(getIntegerVectorAsList[1])
print(getIntegerVectorAsListType)
print(arraysToString(getIntegerVectorAsList))
if(getIntegerVectorAsListType ~= "number") then
	error("getIntegerVectorAsList not Integer!")
end

print("getIntegerMatrix")
if(type(getIntegerMatrix) ~= "nil") then
	error("getIntegerMatrix already defined!")
end
getIntegerMatrix = putIntegerMatrix
getIntegerMatrixType = type(getIntegerMatrix[1][1])
print(getIntegerMatrixType)
print(arraysToString(getIntegerMatrix))
if(getIntegerMatrixType ~= "number") then
	error("getIntegerMatrix not Integer!")
end

print("getIntegerMatrixAsList")
if(type(getIntegerMatrixAsList) ~= "nil") then
	error("getIntegerMatrixAsList already defined!")
end
getIntegerMatrixAsList = putIntegerMatrixAsList
getIntegerMatrixAsListType = type(getIntegerMatrixAsList[1][1])
print(getIntegerMatrixAsListType)
print(arraysToString(getIntegerMatrixAsList))
if(getIntegerMatrixAsListType ~= "number") then
	error("getIntegerMatrixAsList not Integer!")
end
